sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java

```diff
@@ -910,18 +911,29 @@ Mono<VerifyResult> verifyData(SignatureAlgorithm algorithm, byte[] data, byte[]
         });
     }
 
-    private Mono<Boolean> ensureValidKeyAvailable() {
+    private Mono<Boolean> isValidKeyLocallyAvailable() {
+        if (localOperationNotSupported) {
+            return Mono.defer(() -> Mono.just(false));
```

Comment:
nit: no need for the `Mono.defer` here. `Mono.defer` should only be used if the value being set in the `Mono.just` is expensive to compute to defer that computation until the Mono, or Flux, is subscribed. This can just be `Mono.just`.

And actually, in this case, this would make performance worse as `Mono.defer` isn't `Fuseable` which Reactor uses to merge multiple reactive operations into a more performant computation.